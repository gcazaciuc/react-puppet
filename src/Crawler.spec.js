const Crawler = require('./Crawler');
const Printer = require('../src/printers');
const path = require('path');
jest.setTimeout(15000);
const createTest = async (fixture, selector) => {
    const printer = new Printer();
    const crawler = new Crawler();
    const config = {
        slices: [
            {
                url: `file://${path.resolve(fixture)}`,
                sel: selector,
                name: 'Component',
                sheetName: 'ComponentStyle.ts'
            }
        ]
    };
    crawler.setConfig(config);
    const slices = await crawler.crawl();
    const cfg = crawler.getConfig();
    const {
        Component: { styles, jsCode }
    } = printer.print(slices, cfg);
    return { styles, jsCode };
};

describe('Crawler spec', () => {
    it('Should be able to parse a simple HTML file and generate the styles & React comps', async () => {
        const { styles, jsCode } = await createTest('./src/fixtures/simple.html', '.content');
        expect(styles).toMatchSnapshot();
        expect(jsCode).toMatchSnapshot();
    });

    it.only('Should parse pseudo classes styles on elements', async () => {
        const { styles, jsCode } = await createTest(
            './src/fixtures/pseudo-classes.html',
            '.content'
        );
        
        expect(styles).toMatchSnapshot();
        expect(jsCode).toMatchSnapshot();
    });
});
