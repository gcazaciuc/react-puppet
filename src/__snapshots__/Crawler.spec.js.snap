// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Crawler spec Should be able to construct hierarchical slices 1`] = `
"import { style } from 'typestyle';
const colors = { purpleHeart: '#0000FF', fern: '#008000' };
export const content = style({
    width: '100px',
    height: '100px',
    background: '#0000FF',
    $nest: { '&:hover': { background: '#008000' } }
});
"
`;

exports[`Crawler spec Should be able to construct hierarchical slices 2`] = `
"import * as React from 'react';
import * as componentStyle from './ComponentStyle';
export class Component extends React.Component {
    render() {
        return <p className={componentStyle.content}>Test content</p>;
    }
}
"
`;

exports[`Crawler spec Should be able to parse a simple HTML file and generate the styles & React comps 1`] = `
"import { style } from 'typestyle';
const colors = { purpleHeart: '#0000FF', fern: '#008000' };
export const content = style({
    width: '100px',
    height: '100px',
    background: '#0000FF',
    $nest: { '&:hover': { background: '#008000' } }
});
"
`;

exports[`Crawler spec Should be able to parse a simple HTML file and generate the styles & React comps 2`] = `
"import * as React from 'react';
import * as componentStyle from './ComponentStyle';
export class Component extends React.Component {
    render() {
        return <p className={componentStyle.content}>Test content</p>;
    }
}
"
`;

exports[`Crawler spec Should parse inline styles on elements 1`] = `
"import { style } from 'typestyle';
const colors = { purpleHeart: '#0000FF', fern: '#008000' };
export const content = style({
    width: '100px',
    height: '100px',
    background: '#0000FF',
    $nest: { '&:hover': { background: '#008000' } }
});
"
`;

exports[`Crawler spec Should parse inline styles on elements 2`] = `
"import * as React from 'react';
import * as componentStyle from './ComponentStyle';
export class Component extends React.Component {
    render() {
        return <p className={componentStyle.content}>Test content</p>;
    }
}
"
`;

exports[`Crawler spec Should parse pseudo classes styles on elements 1`] = `
"import { style } from 'typestyle';
const colors = { purpleHeart: '#0000FF', fern: '#008000' };
export const content = style({
    width: '100px',
    height: '100px',
    background: '#0000FF',
    $nest: { '&:hover': { background: '#008000' } }
});
"
`;

exports[`Crawler spec Should parse pseudo classes styles on elements 2`] = `
"import * as React from 'react';
import * as componentStyle from './ComponentStyle';
export class Component extends React.Component {
    render() {
        return <p className={componentStyle.content}>Test content</p>;
    }
}
"
`;

exports[`Crawler spec Should transform relative URLs to absolute 1`] = `
"import { style } from 'typestyle';
const colors = {};
export const imgCls = style({});
"
`;

exports[`Crawler spec Should transform relative URLs to absolute 2`] = `
"import * as React from 'react';
import * as componentStyle from './ComponentStyle';
export class Component extends React.Component {
    render() {
        return (
            <p className={componentStyle.imgCls}>
                <img
                    src={
                        'file:///home/gcazaciuc/slice-machine/src/fixtures/absolute-urls.html/img.jpg'
                    }
                    className={componentStyle.imgCls}
                />
            </p>
        );
    }
}
"
`;
